### DISCUSSION TITLE
Minimizing for MSE or maximizing for R^2 gives exactly the same result

### DISCUSSION THREAD

**Discussion by antonkratz** [optuna/optuna#4945] [Q&A] [answered]
I implemented a script with XGBoost regressor and Optuna. (I fix the random seed of TPESampler to suppress randomness). I try two different optimization strategies: minimizing MSE and maximizing R^2.  _With both strategies, I get exactly the same optimized hyperparameters._ Totally confused about this. I expected that if I try to improve a model using these two different kinds of measure, both strategies should arrive at very similar, but definitely not absolutely identical hyperparameters. Yet both give me _exactly_ the same gamma, colsample_bytree, subsample, min_child_weight, max_depth... Is this normal or does it indicate a problem with my script or model?

**Comment by nzw0301** [comment, marked as answer]
From the two metrics' definitions, the optimal predictions are the same; so the observed result seems fine.
